test_wcsapi_extension fails with Python 3.9 on Windows amd64 (Azure)

The following error happens systematically on Azure on Windows 64 with Python 3.9 (while testing the wheels):

```
2021-03-19T11:46:48.0319396Z ================================== FAILURES ===================================
2021-03-19T11:46:48.0320534Z ____________________________ test_wcsapi_extension ____________________________
2021-03-19T11:46:48.0321222Z 
2021-03-19T11:46:48.0322069Z tmpdir = local('C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pytest-of-VssAdministrator\\pytest-0\\test_wcsapi_extension0')
2021-03-19T11:46:48.0322960Z 
2021-03-19T11:46:48.0323803Z     def test_wcsapi_extension(tmpdir):
2021-03-19T11:46:48.0324626Z         # Test that we can build a simple C extension with the astropy.wcs C API
2021-03-19T11:46:48.0325346Z     
2021-03-19T11:46:48.0326002Z         build_dir = tmpdir.mkdir('build').strpath
2021-03-19T11:46:48.0326669Z         install_dir = tmpdir.mkdir('install').strpath
2021-03-19T11:46:48.0327952Z     
2021-03-19T11:46:48.0328649Z         setup_path = os.path.dirname(__file__)
2021-03-19T11:46:48.0329394Z         astropy_path = os.path.abspath(
2021-03-19T11:46:48.0330520Z             os.path.join(setup_path, '..', '..', '..', '..'))
2021-03-19T11:46:48.0331980Z     
2021-03-19T11:46:48.0332774Z         env = os.environ.copy()
2021-03-19T11:46:48.0333498Z         paths = [install_dir, astropy_path]
2021-03-19T11:46:48.0334658Z         if env.get('PYTHONPATH'):
2021-03-19T11:46:48.0335609Z             paths.append(env.get('PYTHONPATH'))
2021-03-19T11:46:48.0336421Z         env['PYTHONPATH'] = os.pathsep.join(paths)
2021-03-19T11:46:48.0338395Z     
2021-03-19T11:46:48.0339533Z         # Build the extension
2021-03-19T11:46:48.0340411Z         # This used to use subprocess.check_call, but on Python 3.4 there was
2021-03-19T11:46:48.0341568Z         # a mysterious Heisenbug causing this to fail with a non-zero exit code
2021-03-19T11:46:48.0342341Z         # *unless* the output is redirected.  This bug also did not occur in an
2021-03-19T11:46:48.0343531Z         # interactive session, so it likely had something to do with pytest's
2021-03-19T11:46:48.0344284Z         # output capture
2021-03-19T11:46:48.0345017Z         p = subprocess.Popen([sys.executable, 'setup.py', 'build',
2021-03-19T11:46:48.0345803Z                               f'--build-base={build_dir}', 'install',
2021-03-19T11:46:48.0346758Z                               f'--install-lib={install_dir}',
2021-03-19T11:46:48.0347536Z                               astropy_path], cwd=setup_path, env=env,
2021-03-19T11:46:48.0348359Z                               stdout=subprocess.PIPE, stderr=subprocess.PIPE)
2021-03-19T11:46:48.0349065Z     
2021-03-19T11:46:48.0349723Z         # Whether the process fails or not this isn't likely to produce a great
2021-03-19T11:46:48.0350587Z         # deal of output so communicate should be fine in almost all cases
2021-03-19T11:46:48.0351323Z         stdout, stderr = p.communicate()
2021-03-19T11:46:48.0351976Z     
2021-03-19T11:46:48.0352654Z         try:
2021-03-19T11:46:48.0353383Z             stdout, stderr = stdout.decode('utf8'), stderr.decode('utf8')
2021-03-19T11:46:48.0354125Z         except UnicodeDecodeError:
2021-03-19T11:46:48.0354812Z             # Don't try to guess about encoding; just display the text
2021-03-19T11:46:48.0355595Z             stdout, stderr = stdout.decode('latin1'), stderr.decode('latin1')
2021-03-19T11:46:48.0356326Z     
2021-03-19T11:46:48.0357017Z         # If compilation fails, we can skip this test, since the
2021-03-19T11:46:48.0357789Z         # dependencies necessary to compile an extension may be missing.
2021-03-19T11:46:48.0358570Z         # If it passes, however, we want to continue and ensure that the
2021-03-19T11:46:48.0359520Z         # extension created is actually usable.  However, if we're on
2021-03-19T11:46:48.0360878Z         # continuous integration setup, we
2021-03-19T11:46:48.0361743Z         # don't want to ever skip, because having it fail in that
2021-03-19T11:46:48.0362595Z         # environment probably indicates something more serious that we
2021-03-19T11:46:48.0364101Z         # want to know about.
2021-03-19T11:46:48.0371227Z         if (not ('CI' in os.environ or
2021-03-19T11:46:48.0372171Z                  'CONTINUOUS_INTEGRATION' in os.environ) and
2021-03-19T11:46:48.0373032Z                 p.returncode):
2021-03-19T11:46:48.0373817Z             pytest.skip("system unable to compile extensions")
2021-03-19T11:46:48.0374558Z             return
2021-03-19T11:46:48.0375569Z     
2021-03-19T11:46:48.0376239Z >       assert p.returncode == 0, (
2021-03-19T11:46:48.0377297Z             "setup.py exited with non-zero return code {}\n"
2021-03-19T11:46:48.0378394Z             "stdout:\n\n{}\n\nstderr:\n\n{}\n".format(
2021-03-19T11:46:48.0379275Z                 p.returncode, stdout, stderr))
2021-03-19T11:46:48.0380610Z E       AssertionError: setup.py exited with non-zero return code 1
2021-03-19T11:46:48.0381583Z E         stdout:
2021-03-19T11:46:48.0382286Z E         
2021-03-19T11:46:48.0383155Z E         running build
2021-03-19T11:46:48.0383690Z 
2021-03-19T11:46:48.0384358Z E         running build_ext
2021-03-19T11:46:48.0384836Z 
2021-03-19T11:46:48.0385606Z E         No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
2021-03-19T11:46:48.0386513Z 
2021-03-19T11:46:48.0387201Z E         building 'wcsapi_test' extension
2021-03-19T11:46:48.0387851Z 
2021-03-19T11:46:48.0388675Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9
2021-03-19T11:46:48.0389306Z 
2021-03-19T11:46:48.0390136Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release
2021-03-19T11:46:48.0390777Z 
2021-03-19T11:46:48.0391619Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users
2021-03-19T11:46:48.0392325Z 
2021-03-19T11:46:48.0393264Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator
2021-03-19T11:46:48.0393998Z 
2021-03-19T11:46:48.0395497Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata
2021-03-19T11:46:48.0396347Z 
2021-03-19T11:46:48.0397207Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local
2021-03-19T11:46:48.0397918Z 
2021-03-19T11:46:48.0398825Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp
2021-03-19T11:46:48.0399487Z 
2021-03-19T11:46:48.0400370Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk
2021-03-19T11:46:48.0401101Z 
2021-03-19T11:46:48.0401978Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib
2021-03-19T11:46:48.0403414Z 
2021-03-19T11:46:48.0404379Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages
2021-03-19T11:46:48.0405144Z 
2021-03-19T11:46:48.0406267Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy
2021-03-19T11:46:48.0407051Z 
2021-03-19T11:46:48.0407955Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs
2021-03-19T11:46:48.0408692Z 
2021-03-19T11:46:48.0410012Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests
2021-03-19T11:46:48.0410842Z 
2021-03-19T11:46:48.0411841Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension
2021-03-19T11:46:48.0412801Z 
2021-03-19T11:46:48.0416769Z E         C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DYY_NO_UNISTD_H -D_CRT_SECURE_NO_WARNINGS -D_NO_OLDNAMES -DNO_OLDNAMES -D__STDC__ -Ic:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\numpy\core\include -Ic:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\include\astropy_wcs -Ic:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\include\wcslib -I/usr/include\wcslib -Ic:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\include -IC:\cibw\python\python.3.9.2\tools\include -IC:\cibw\python\python.3.9.2\tools\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\ATLMFC\include -IC:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\include -IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt -IC:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt /Tcc:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.c /FoC:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.obj
2021-03-19T11:46:48.0420702Z 
2021-03-19T11:46:48.0421883Z E         creating C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\lib.win-amd64-3.9
2021-03-19T11:46:48.0422743Z 
2021-03-19T11:46:48.0533055Z E         C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\libs /LIBPATH:C:\cibw\python\python.3.9.2\tools\libs /LIBPATH:C:\cibw\python\python.3.9.2\tools /LIBPATH:c:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\PCbuild\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\lib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64 /EXPORT:PyInit_wcsapi_test C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.obj /OUT:C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\lib.win-amd64-3.9\wcsapi_test.cp39-win_amd64.pyd /IMPLIB:C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.cp39-win_amd64.lib
2021-03-19T11:46:48.0537914Z 
2021-03-19T11:46:48.0539168Z E         LINK : fatal error LNK1104: cannot open file 'C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.cp39-win_amd64.lib'
2021-03-19T11:46:48.0540660Z 
2021-03-19T11:46:48.0541371Z E         
2021-03-19T11:46:48.0542173Z E         
2021-03-19T11:46:48.0542935Z E         stderr:
2021-03-19T11:46:48.0543629Z E         
2021-03-19T11:46:48.0546916Z E         error: Command "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\libs /LIBPATH:C:\cibw\python\python.3.9.2\tools\libs /LIBPATH:C:\cibw\python\python.3.9.2\tools /LIBPATH:c:\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\PCbuild\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\lib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64 /EXPORT:PyInit_wcsapi_test C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.obj /OUT:C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\lib.win-amd64-3.9\wcsapi_test.cp39-win_amd64.pyd /IMPLIB:C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_wcsapi_extension0\build\temp.win-amd64-3.9\Release\users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\wcsapi_test.cp39-win_amd64.lib" failed with exit status 1104
2021-03-19T11:46:48.0550661Z 
2021-03-19T11:46:48.0551581Z E         
2021-03-19T11:46:48.0552280Z E         
2021-03-19T11:46:48.0553807Z E       assert 1 == 0
2021-03-19T11:46:48.0554849Z E        +  where 1 = <Popen: returncode: 1 args: ['c:\\users\\vssadministrator\\appdata\\local\\t...>.returncode
2021-03-19T11:46:48.0555673Z 
2021-03-19T11:46:48.0556908Z users\vssadministrator\appdata\local\temp\tmp7qbupvdk\lib\site-packages\astropy\wcs\tests\extension\test_extension.py:63: AssertionError
```

The issue is that one of the paths is too long by a few characters - if I rename the test to something shorter it works. However I'm not convinced that's the best fix, so I'm curious if anyone has better ideas to avoid the really long paths?

In the mean time we can just skip this test on Azure (PR forthcoming)

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -18,7 +18,7 @@ jobs:
       upload_to_pypi: false
       upload_to_anaconda: false
       test_extras: test
-      test_command: pytest -p no:warnings --astropy-header -m "not hypothesis" -k "not test_data_out_of_range and not test_wcsapi_extension and not test_set_locale and not TestQuantityTyping" --pyargs astropy
+      test_command: pytest -p no:warnings --astropy-header -m "not hypothesis" -k "not test_data_out_of_range and not test_set_locale and not TestQuantityTyping" --pyargs astropy
       targets: |
         - cp39-manylinux_x86_64
 
@@ -39,7 +39,7 @@ jobs:
       # currently fails, see https://github.com/astropy/astropy/issues/10409
       # We also exclude test_set_locale as it sometimes relies on the correct locale
       # packages being installed, which it isn't always.
-      test_command: pytest -p no:warnings --astropy-header -m "not hypothesis" -k "not test_data_out_of_range and not test_wcsapi_extension and not test_set_locale and not TestQuantityTyping" --pyargs astropy
+      test_command: pytest -p no:warnings --astropy-header -m "not hypothesis" -k "not test_data_out_of_range and not test_set_locale and not TestQuantityTyping" --pyargs astropy
       targets: |
         # Linux wheels
         - cp38-manylinux_i686

